modules/samples/firealarm/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.firealarm.type-view/public/js/download.js

```diff
@@ -0,0 +1,195 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var modalPopup = ".wr-modalpopup";
+var modalPopupContainer = modalPopup + " .modalpopup-container";
+var modalPopupContent = modalPopup + " .modalpopup-content";
+var body = "body";
+
+/*
+ * set popup maximum height function.
+ */
+function setPopupMaxHeight() {
+    $(modalPopupContent).css('max-height', ($(body).height() - ($(body).height() / 100 * 30)));
+    $(modalPopupContainer).css('margin-top', (-($(modalPopupContainer).height() / 2)));
+}
+
+/*
+ * show popup function.
+ */
+function showPopup() {
+    $(modalPopup).show();
+    setPopupMaxHeight();
+    $('#downloadForm').validate({
+                                    rules: {
+                                        deviceName: {
+                                            minlength: 4,
+                                            required: true
+                                        }
+                                    },
+                                    highlight: function (element) {
+                                        $(element).closest('.control-group').removeClass('success').addClass('error');
+                                    },
+                                    success: function (element) {
+                                        $(element).closest('.control-group').removeClass('error').addClass('success');
+                                        $('label[for=deviceName]').remove();
+                                    }
+                                });
+    var deviceType = "";
+    $('.deviceType').each(function () {
+        if (this.value != "") {
+            deviceType = this.value;
+        }
+    });
+}
+
+/*
+ * hide popup function.
+ */
+function hidePopup() {
+    $('label[for=deviceName]').remove();
+    $('.control-group').removeClass('success').removeClass('error');
+    $(modalPopupContent).html('');
+    $(modalPopup).hide();
+}
+
+/*
+ * DOM ready functions.
+ */
+$(document).ready(function () {
+    attachEvents();
+});
+
+function attachEvents() {
+    /**
+     * Following click function would execute
+     * when a user clicks on "Download" link
+     * on Device Management page in WSO2 DC Console.
+     */
+    $("a.download-link").click(function () {
+        var sketchType = $(this).data("sketchtype");
+        var deviceType = $(this).data("devicetype");
+        var downloadDeviceAPI = "/devicemgt/api/devices/sketch/generate_link";
+        var payload = {"sketchType": sketchType, "deviceType": deviceType};
+        $(modalPopupContent).html($('#download-device-modal-content').html());
+        showPopup();
+        var deviceName;
+        $("a#download-device-download-link").click(function () {
+            $('.new-device-name').each(function () {
+                if (this.value != "") {
+                    deviceName = this.value;
+                }
+            });
+            $('label[for=deviceName]').remove();
```

Comment:
Be consistent on using quotes for string definitions. use double quotes as the standard.